confignode/src/main/java/org/apache/iotdb/confignode/client/async/handlers/FetchSchemaBlackLsitHandler.java

```diff
@@ -0,0 +1,101 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.confignode.client.async.handlers;
+
+import org.apache.iotdb.common.rpc.thrift.TDataNodeLocation;
+import org.apache.iotdb.common.rpc.thrift.TSStatus;
+import org.apache.iotdb.confignode.client.DataNodeRequestType;
+import org.apache.iotdb.mpp.rpc.thrift.TFetchSchemaBlackListResp;
+import org.apache.iotdb.rpc.RpcUtils;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import org.apache.thrift.async.AsyncMethodCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+public class FetchSchemaBlackLsitHandler extends AbstractRetryHandler
+    implements AsyncMethodCallback<TFetchSchemaBlackListResp> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FetchSchemaBlackLsitHandler.class);
+
+  private Map<Integer, TFetchSchemaBlackListResp> dataNodeResponseMap = new HashMap<>();
+
+  public FetchSchemaBlackLsitHandler(Map<Integer, TDataNodeLocation> dataNodeLocationMap) {
+    super(DataNodeRequestType.FETCH_SCHEMA_BLACK_LIST, dataNodeLocationMap);
+  }
+
+  public FetchSchemaBlackLsitHandler(
+      CountDownLatch countDownLatch,
+      TDataNodeLocation targetDataNode,
+      Map<Integer, TDataNodeLocation> dataNodeLocationMap) {
+    super(
+        countDownLatch,
+        DataNodeRequestType.FETCH_SCHEMA_BLACK_LIST,
+        targetDataNode,
+        dataNodeLocationMap);
+  }
+
+  public Map<Integer, TFetchSchemaBlackListResp> getDataNodeResponseMap() {
+    return dataNodeResponseMap;
+  }
+
+  @Override
+  public void onComplete(TFetchSchemaBlackListResp tFetchSchemaBlackListResp) {
+    TSStatus tsStatus = tFetchSchemaBlackListResp.getStatus();
+    dataNodeResponseMap.put(targetDataNode.getDataNodeId(), tFetchSchemaBlackListResp);
+    if (tsStatus.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      dataNodeLocationMap.remove(targetDataNode.getDataNodeId());
+      LOGGER.info("Successfully fetch schema black list on DataNode: {}", targetDataNode);
+    } else if (tsStatus.getCode() == TSStatusCode.MULTIPLE_ERROR.getStatusCode()) {
+      dataNodeLocationMap.remove(targetDataNode.getDataNodeId());
+      LOGGER.error(
+          "Failed to fetch schema black list on DataNode {}, {}",
+          dataNodeLocationMap.get(targetDataNode.getDataNodeId()),
+          tsStatus);
```

Comment:
You'll get a NPE if you remove before get T^T

Reply:
Fixed.